asset-transfer-private-data/chaincode-go/privateAssetTransfer.go

```diff
@@ -0,0 +1,469 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// Asset describes main asset details that are visible to all organizations
+type Asset struct {
+	ObjectType 		 string `json:"object_type"`  //docType is used to distinguish the various types of objects in state database
```

Comment:
favoring simplicity: 
```suggestion
	Type 		 string `json:"type"`  //Type can be used to distinguish various Asset types in your business logic
```